backend/corpora/gale/gale.py

```diff
@@ -0,0 +1,309 @@
+import re
+from datetime import datetime
+import openpyxl
+import os.path
+import PIL
+
+from ianalyzer_readers.xml_tag import Tag, SiblingTag, ParentTag
+from ianalyzer_readers import extract
+
+from addcorpus.python_corpora import filters
+from addcorpus.python_corpora.corpus import XMLCorpusDefinition, FieldDefinition
+from addcorpus.es_mappings import keyword_mapping, main_content_mapping
+from addcorpus.es_settings import es_settings
+from api.utils import find_media_file
+from media.media_url import media_url
+
+
+def fix_path_sep(path):
+    return path.replace('\\', os.path.sep).lstrip(os.path.sep).strip()
+
+
+class GaleCorpus(XMLCorpusDefinition):
+    @property
+    def es_settings(self):
+        return es_settings(self.languages[:1], stopword_analysis=True, stemming_analysis=True)
+
+    tag_toplevel = Tag('articles')
+    tag_entry = Tag('artInfo')
+    external_file_tag_toplevel = Tag('issue')
+
+    scan_image_type = 'image/png'
+
+    def sources(self, start, end):
+        filename, sheet = self.metafile
+        wb = openpyxl.load_workbook(filename=filename)
+        for index, row in enumerate(wb[sheet].values):
+            # skip first row, and rows without date
+            if index==0 or not row[1]:
+                continue
+
+            publication_tile, issue_date, image_location, data_location, filename = row
+
+            data_location = fix_path_sep(data_location)
+            image_path = fix_path_sep(image_location)
+
+            if issue_date.startswith('[') and issue_date.endswith(']'):
+                issue_date = issue_date[1:-1]
+
+            if issue_date == 'Date Unknown':
+                date_full = None
+            else:
+                date_full = datetime.strptime(issue_date, '%B %d, %Y').strftime('%Y-%m-%d')
+
+            issue_id = filename.split("_")[0]
+            external_file = os.path.join(self.data_directory, data_location, filename)
+            xml_file = os.path.join(self.data_directory, data_location, issue_id + '_Text.xml')
+            if not os.path.isfile(xml_file):
+                print("File {} not found".format(xml_file))
+                continue
+
+            meta = dict(title=publication_tile, image_path=image_path, issue_id=issue_id, external_file=external_file,
+                        date_full=date_full)
+            yield xml_file, meta
+
+    def process_scan(self, src):
+        dst = src
+        base, ext = os.path.splitext(src)
+        if ext.lower() == '.tif':
+            dst = base + '.png'
+            if not os.path.isfile(dst):
+                PIL.Image.open(src).save(dst)
+        return dst
```

Comment:
As I understand, a single TIFF image can contain multiple images (e.g. a multi-page article). Just to double check that that is not the case here.